craft_application/pytest_plugin.py

```diff
@@ -33,14 +34,35 @@
     from pyfakefs.fake_filesystem import FakeFilesystem
 
 
+@pytest.fixture(autouse=True, scope="session")
+def reset_craft_environment() -> Iterator[None]:
+    """Reset any relevant environment variables during testing.
+
+    This clears any ``CRAFT_*`` or ``SNAP_*`` environment variables that are set
+    in the environment. To keep an environment variable while running tests for
+    debugging purposes, put its name in the environment variable
+    ``CRAFT_DEBUG_KEEP_ENV_VARS``, which is a comma-separated list of variables.
+    """
+    keep_vars = set(os.environ.get("CRAFT_DEBUG_KEEP_ENV_VARS", "").split())
```

Comment:
```suggestion
    keep_vars = set(os.environ.get("CRAFT_DEBUG_KEEP_ENV_VARS", "").split(","))
```
The docstring says to use commas, but `.split()` defaults to whitespace